tehom/downloads.py

```diff
@@ -80,8 +87,72 @@ def _download_ais_to_temp(year: int, month: int, zone: int) -> Path:
     Returns:
         location of download result
     """
-    # morgan
-    pass
+    # MWM, 07/23/2021: Done; predownload logic parses filename
+
+    if month < 10:
+        month = f"0{month}"
+    else:
+        month = str(month)
+
+    if year > 2017:
+        if zone < 10:
+            zone = f"0{zone}"
+        url = (
+            "https://coast.noaa.gov/htdata/CMSP/AISDataHandler/"
+            f"{year}/AIS_{year}_{month}_{zone}.zip"
+        )
+    elif year > 2014:
+        if zone < 10:
+            zone = f"0{zone}"
+        url = (
+            "https://coast.noaa.gov/htdata/CMSP/AISDataHandler/"
+            f"{year}/AIS_{year}_{month}_Zone{zone}.zip"
+        )
+    elif year == 2014:
+        url = (
+            "https://coast.noaa.gov/htdata/CMSP/AISDataHandler/"
+            f"{year}/{month}/Zone{zone}_{year}_{month}.zip"
+        )
+    elif year > 2010:
+        url = (
+            "https://coast.noaa.gov/htdata/CMSP/AISDataHandler/"
+            f"{year}/{month}/Zone{zone}_{year}_{month}.gdb.zip"
+        )
+    elif year < 2010:
+        months = {
+            "01": "January",
+            "02": "February",
+            "03": "March",
+            "04": "April",
+            "05": "May",
+            "06": "June",
+            "07": "July",
+            "08": "August",
+            "09": "September",
+            "10": "October",
+            "11": "November",
+            "12": "December",
+        }
+        url = (
+            "https://coast.noaa.gov/htdata/CMSP/AISDataHandler/{year}/"
+            f"{month}_{months[month]}_{year}/Zone{zone}_{year}_{month}.zip"
+        )
+
+    if isinstance(zone, int) and zone < 10:
+        zone = f"0{zone}"
+    filepath = _persistence.AIS_TEMP_DIR / f"{year}_{month}_{zone}.zip"
+    if not filepath.exists():
+        try:
+            print(f"Downloading data for {year} {month}, Zone {zone}...")
```

Comment:
I'm not a fan of leaving in diagnostic print statements.  If they're potentially worth including, add 
`logger = logging.getLogger(__name__)` at the beginning of the module, after import statements.  Then, instead of `print(...)`, use `logger.info(...)`.

This allows users (or applications using this library) to suppress or redirect outputs to a log.